### PR TITLE
[CFE-483]: Feat(pages) - update my publish module to support lcd endpoint

### PR DESCRIPTION
**Pull request by Poafs1**

### CODE REVIEW

File: src/lib/services/move/module/index.ts [hunk 1]
```diff
@@ -74,25 +73,27 @@ export const useModuleByAddressLcd = ({
...
-  address: Addr;
-  options?: Omit<UseQueryOptions<IndexedModule[]>, "queryKey">;
+  address: Option<Addr>;
+  options?: Omit<UseQueryOptions<{ items: IndexedModule[] }>, "queryKey">;
 }) => {
-  const baseEndpoint = useBaseApiRoute("rest");
-  const queryFn = () => getModulesByAddressLcd(baseEndpoint, address);
+  const lcdEndpoint = useLcdEndpoint();
+  const queryFn = () => {
+    if (!address) throw new Error("address is undefined");
```
Comment (evilpeach) [inline src/lib/services/move/module/index.ts:83]: `address is undefined (useModulesByAddressLcd)`

File: src/lib/services/move/module/lcd.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { parseWithError } from "lib/utils";
+
+export const getModuleByAddressLcd = async (
+  baseEndpoint: string,
+  address: Addr,
+  moduleName: string
+): Promise<IndexedModule> =>
+  axios
+    .get(
+      `${baseEndpoint}/initia/move/v1/accounts/${encodeURIComponent(address)}/modules/${encodeURIComponent(moduleName)}`
```
Comment (evilpeach) [inline src/lib/services/move/module/lcd.ts:14]: If address is string, I think it's okm to use `encodeURI`.

File: src/lib/pages/my-published-modules/components/MyPublishedModulesTable.tsx [hunk 3]
```diff
@@ -1,25 +1,32 @@
...
+import { useModulesByAddress, useModulesByAddressLcd } from "lib/services/move";
 import type { ModuleInfo, Option } from "lib/types";
 
 export const MyPublishedModulesTable = () => {
   const [keyword, setKeyword] = useState("");
   const { address } = useCurrentChain();
+  const isFullTier = useTierConfig() === "full";
+
+  const fullData = useModulesByAddress(address, isFullTier);
+  const liteData = useModulesByAddressLcd({
```
Comment (songwongtp) [inline src/lib/pages/my-published-modules/components/MyPublishedModulesTable.tsx:20]: resApi, resLcd

File: src/lib/services/move/module/index.ts [hunk 4]
```diff
@@ -74,25 +73,28 @@ export const useModuleByAddressLcd = ({
...
+  const queryFn = () => {
+    if (!address)
+      throw new Error("address is undefined (useModulesByAddressLcd)");
+    return getModulesByAddressLcd(lcdEndpoint, address);
+  };
 
-  return useQuery<IndexedModule[]>(
-    [CELATONE_QUERY_KEYS.ACCOUNT_MODULES, baseEndpoint, address],
+  return useQuery<{ items: IndexedModule[] }>(
+    [CELATONE_QUERY_KEYS.ACCOUNT_MODULES, lcdEndpoint, address],
```
Comment (songwongtp) [inline src/lib/services/move/module/index.ts:89]: CELATONE_QUERY_KEYS.MODULES_BY_ADDRESS_LCD

**Review by songwongtp — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Review by songwongtp — APPROVED** [review]

**Pull request merged**